corehq/apps/accounting/subscription_changes.py

```diff
@@ -91,13 +92,19 @@ def privilege_to_response_function(cls):
             privileges.COMMCARE_LOGO_UPLOADER: cls.response_commcare_logo_uploader,
             privileges.ADVANCED_DOMAIN_SECURITY: cls.response_domain_security,
             privileges.PRACTICE_MOBILE_WORKERS: cls.response_practice_mobile_workers,
+            privileges.MOBILE_WORKER_CREATION: cls.response_mobile_worker_creation,
         }
         privs_to_responses.update({
             p: cls.response_report_builder
             for p in privileges.REPORT_BUILDER_ADD_ON_PRIVS
         })
         return privs_to_responses
 
+    def get_response(self):
```

Comment:
@nickpell `get_response` sometimes returns a boolean (used [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/accounting/models.py#L961-L962) to determine whether the upgrade/downgrade was successful), and other times returns a list of messages alerting the user about the upgrade/downgrade (used [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/views.py#L1566-L1566)).  It looks like the presence of messages (or a boolean `True` value doesn't necessaily indicate that the upgrade/downgrade failed though.

Anyways, I find it a bit confusing, do you have any suggestions for making more sense of this?

Reply:
This is confusing - what happens is, any ```None``` return values get filtered out, and then something happens with the non-```None``` return values.  For ```BaseModifySubscriptionActionHandler``` this means taking ```all(response)```, for ```DomainDowngradeStatusHandler``` it means just returning the responses.

A better approach (probably out of scope for this PR) is to have each function return a non-```None``` value explicitly.  For now returning ```True``` when the function succeeds is good enough.